backend/inspirehep/orcid/domain_models.py

```diff
@@ -125,42 +125,56 @@ def push(self):
             putcode = self._post_or_put_work(putcode)
         except orcid_client_exceptions.WorkAlreadyExistsException:
             # We POSTed the record as new work, but it failed because the work
-            # already exists (identified by the external identifiers).
+            # with the same identifier is already in ORCID.
             # This means we do not have the putcode, thus we cache all
-            # author's putcodes and PUT the work again.
-            try:
-                if self.do_fail_if_duplicated_identifier:
+            # author's putcodes. If the putcode wasn't found we try to push
+            # once again works with clashing identifier to update them and
+            # resolve conflict.
+            if self.pushing_duplicated_identifier:
+                # In case the WorkAlreadyExistsException
+                # exception was obtained when pushing records
+                # with clashing identifiers (what should never happen)
+                # DuplicatedExternalIdentifierPusherException is raised.
```

Comment:
I don't think this comment is helpful, you simply repeat what is going on (which can be understood by reading the code)